### ISSUE TITLE
Is there any purpose to allow_any_origin()?

### ISSUE THREAD

**Issue by Newbytee** [seanmonstar/warp#614]
Greetings. 

In my efforts to learn warp, I'm experimenting `warp::cors()` and I'm seemingly finding that it implicitly allows any origins **without** specifying `allow_any_origin()`. Curious, I gave removing `allow_any_origin()` on line 59 in the cors.rs test a go, and all tests passed still.

Is this intended, and if so, what is the purpose of `allow_any_origin()`?

**Comment by ernestas-poskus** [comment]
> allow_any_origin()

Method simulates https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin and allows all origins, though does explicitly insert ACAO header with `*` asterisk value. 

> A response that tells the browser to allow code from any origin to access a resource will include the following:
> Access-Control-Allow-Origin: *

**Comment by Newbytee** [comment]
> Method simulates https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin and allows all origins, though does explicitly insert ACAO header with `*` asterisk value.

Yes, that's how I imagined it would work, but it's actually not necessary for this. Here is its implementation: https://github.com/seanmonstar/warp/blob/master/src/filters/cors.rs#L193

Notice that it sets `self.origins` to None. Now, what is the default value of `self.origins`? Line 47 tells us: https://github.com/seanmonstar/warp/blob/master/src/filters/cors.rs#L47

It's also None. Therefore, I'm asking what the purpose of this is, and when it might be useful. The docs do not mention this behaviour and rather, by how I read it, imply that `allow_any_origin()` is mandatory: https://docs.rs/warp/0.2.3/warp/filters/cors/fn.cors.html
